src/main/webapp/app/guided-tour/guided-tour.service.ts

```diff
@@ -872,19 +872,23 @@ export class GuidedTourService {
     /**
      *  @return true if highlighted element is available, otherwise false
      */
-    private checkSelectorValidity(): boolean {
+    private checkSelectorValidity(): boolean | void {
         if (!this.currentTour) {
             return false;
         }
         const currentTourStep = this.currentTour.steps[this.currentTourStepIndex];
-        const selector = currentTourStep.highlightSelector;
-        if (selector) {
-            const selectedElement = document.querySelector(selector);
-            if (!selectedElement) {
-                return false;
+        if (currentTourStep) {
+            const selector = currentTourStep.highlightSelector ? currentTourStep.highlightSelector : undefined;
+            if (selector) {
+                const selectedElement = document.querySelector(selector);
+                if (!selectedElement) {
+                    return false;
+                }
```

Comment:
Could we have less nested ifs in order to increase code readability?

Reply:
ah, okay.. i'll try the case again on test server after returning false. the tour itself is really annoying..